Add dynamic column generation for UnifiedLogs table

MildTomato opened the pull request:
This PR enables dynamic column generation for the Unified Logs table, improving maintainability and flexibility. Key changes:

Dynamic Columns
	•	Added generateDynamicColumns in Columns.tsx to build columns based on data.
	•	Columns like method, pathname, and event_message are hidden if empty.
	•	Replaced static UNIFIED_LOGS_COLUMNS with dynamic version using useMemo in UnifiedLogs.tsx.

Simplified Definitions
	•	Removed unused metadata and class names from columns.
	•	Standardized column sizing and sorting behavior.

Fallback
	•	Added static fallback using generateDynamicColumns([]) when no data exists.

MildTomato committed:
Remove 'host' field from UnifiedLogs interface

Eliminated the 'host' field from filter fields, schema definitions, and column configuration in UnifiedLogs. This streamlines the logs interface and removes unused or unnecessary data.

MildTomato committed:
Add dynamic column generation for UnifiedLogs table

Replaces static column definitions with a dynamic generator that hides columns when all their values are empty. The UnifiedLogs table now adapts its columns based on the current data, improving readability and relevance.

MildTomato commented on apps/studio/components/interfaces/UnifiedLogs/UnifiedLogs.tsx:
```diff
@@ -165,9 +165,14 @@ export const UnifiedLogs = () => {
     return cn(levelClassName, isPast ? 'opacity-50' : 'opacity-100', 'h-[30px]')
   }
 
+  // Generate dynamic columns based on current data
+  const dynamicColumns = useMemo(() => {
+    return generateDynamicColumns(flatData)
+  }, [flatData])
+
```
@alaister hmm not sure this is right

alaister replied:
I think it's okay

MildTomato committed:
Update Columns.tsx

MildTomato committed:
Update Columns.tsx

MildTomato committed:
Add column width classes to UnifiedLogs columns

Added explicit cell and header class names for column widths in the UnifiedLogs table to ensure consistent sizing and improve layout control.

MildTomato committed:
Update Columns.tsx

alaister reviewed (approved):
LGTM!

- Tested on preview

alaister reviewed (changes requested):
blocking

MildTomato committed:
Merge branch 'master' into chore/remove-host

MildTomato committed:
Merge branch 'master' into feat/dynamic-columns-logs

MildTomato committed:
Merge branch 'chore/remove-host' into feat/dynamic-columns-logs

MildTomato committed:
Refactor dynamic columns to include visibility state

The dynamic columns generator now returns both columns and their visibility state, allowing UnifiedLogs to merge dynamic visibility with existing column visibility. This improves control over which columns are shown based on the data.

MildTomato committed:
Merge branch 'master' into feat/dynamic-columns-logs

MildTomato committed:
Merge branch 'master' into feat/dynamic-columns-logs

alaister reviewed (approved):
LGTM once again!
